File: programmers/타겟넘버.js
function solution(numbers, target) {
  const candidateNumbers = []
  const find = (numbers, idx) => {
    const a = [...numbers]
    const b = [...numbers]
    b[idx] = -b[idx]
    if(numbers.length - 1 === idx) {
      candidateNumbers.push(b)
      candidateNumbers.push(a)
      return 
    }
    find(a, idx + 1)
    find(b, idx + 1)    
  }
  find(numbers, 0)
  return candidateNumbers.reduce((count, candidateNumber) => {
    const sum = candidateNumber.reduce((acc, curr) => acc + curr, 0)
    return count + Number(sum === target)
  }, 0)
}

console.log(solution([1,1,1,1,1], 3))

File: programmers/타겟넘버.py
from collections import deque

def solution(numbers, target):
    firstNumber = numbers[0]
    candidates = [[firstNumber], [-firstNumber]]

    for number in numbers[1:]:
        next_candidates = []
        for candidate in candidates:
            candidate1 = candidate[:]
            candidate2 = candidate[:]
            candidate1.append(number)
            candidate2.append(-number)
            next_candidates.append(candidate1)
            next_candidates.append(candidate2)
        candidates = next_candidates
    
    result = 0
    for candidate in candidates:
        if sum(candidate) == target:
            result += 1
    return result

print(solution([1,2,3,4,5],3))